fog/sql_recovery_db/src/lib.rs

```diff
@@ -935,6 +935,78 @@ impl SqlRecoveryDb {
         }
     }
 
+    /// Get ETxOutRecords for a given ingress key from a block, and subsequent
+    /// blocks, up to some limit. (This is like a batch call to
+    /// get_tx_outs_by_block_and_key_retriable with lookahead, and makes
+    /// sense if there is high network latency.)
+    ///
+    /// Arguments:
+    /// * ingress_key: The ingress key we need ETxOutRecords from
+    /// * block_index: The first block we need ETxOutRecords from
+    /// * block_count: How many subsequent blocks to also request data for.
+    ///
+    /// Returns:
+    /// * The sequence of ETxOutRecord's, from consequecutive blocks starting
+    ///   from block_index. Empty if not even the block_index'th block exists.
+    fn get_tx_outs_by_block_range_and_key_retriable(
+        &self,
+        ingress_key: CompressedRistrettoPublic,
+        block_index: u64,
+        block_count: usize,
+    ) -> Result<Vec<Vec<ETxOutRecord>>, Error> {
+        let conn = self.pool.get()?;
+
+        // The idea is:
+        // Similar to get_tx_outs_by_block_and_key_retriable, but now
+        // * we have a range of admissible block indices
+        // * we order by the block number
+        // * we also select over the block number so that sql gives us the block number
+        //
+        // This ensures that we can detect any gaps in the data
+        let key_bytes: &[u8] = ingress_key.as_ref();
+        let query = {
+            use schema::ingested_blocks::dsl;
+            dsl::ingested_blocks
+                .filter(dsl::ingress_public_key.eq(key_bytes))
+                .filter(dsl::block_number.ge(block_index as i64))
+                .limit(block_count as i64)
+                .select((dsl::block_number, dsl::proto_ingested_block_data))
+                .order(dsl::block_number.asc())
+        };
+
+        // We will get one row for each hit in the table we found
+        let rows: Vec<(i64, Vec<u8>)> = query.load::<(i64, Vec<u8>)>(&conn)?;
```

Comment:
IIUC we don't need to specify the type twice
```suggestion
        let rows: Vec<(i64, Vec<u8>)> = query.load(&conn)?;
```